### PR TITLE
Bug Fix 507364 : Improve Publish Build Artifact related tasks error w…

### PR DESCRIPTION
**Pull request by vishnugms**
Bug Fix: Improve Publish Build Artifact related tasks error when run within Release
https://mseng.visualstudio.com/DefaultCollection/VSOnline/_workitems?id=507364

### CODE REVIEW

File: Tasks/CopyPublishBuildArtifacts/CopyPublishBuildArtifacts.ps1 [hunk 1]
```diff
@@ -28,6 +28,13 @@ Write-Verbose "TargetPath = $TargetPath"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 
+$hostType = Get-TaskVariable $distributedTaskContext "system.hosttype"
+
+if($hostType -ine "build" )
```
Comment (ericsciple) [inline Tasks/CopyPublishBuildArtifacts/CopyPublishBuildArtifacts.ps1]: Not a mandatory change, but wouldn't `-ne 'release'` make more sense? If so then all messages would need to change as well.

File: Tasks/PublishBuildArtifacts/publishbuildartifacts.ts [hunk 2]
```diff
@@ -5,6 +5,14 @@ import tl = require('vsts-task-lib/task');
```
Comment (ericsciple) [inline Tasks/PublishBuildArtifacts/publishbuildartifacts.ts]: When you build, isn't a .js file supposed to be generated as well? Please commit that as well if you forgot.

File: Tasks/PublishBuildArtifacts/publishbuildartifacts.ts [hunk 3]
```diff
@@ -5,6 +5,14 @@ import tl = require('vsts-task-lib/task');
 
 tl.setResourcePath(path.join(__dirname, 'task.json'));
 
+// Getting host type, throwing error if it is invoked through Release
+var hostType: string = tl.getVariable('system.hosttype');
+if(hostType !== "build") {  
+    tl.setResult(tl.TaskResult.Failed, tl.loc('PublishBuildArtifactsNotSupported'));
+} 
+
+tl.warning('Artifact type is not specified.');
```
Comment (ericsciple) [inline Tasks/PublishBuildArtifacts/publishbuildartifacts.ts]: Why is this line here?

File: Tasks/PublishBuildArtifacts/publishbuildartifacts.ts [hunk 4]
```diff
@@ -5,6 +5,14 @@ import tl = require('vsts-task-lib/task');
 
 tl.setResourcePath(path.join(__dirname, 'task.json'));
 
+// Getting host type, throwing error if it is invoked through Release
```
Comment (ericsciple) [inline Tasks/PublishBuildArtifacts/publishbuildartifacts.ts]: This task has L0's that should be kept up to date. That could be a follow-up change so this doesn't miss sprint cut-off. But they should be updated.

File: Tasks/CopyPublishBuildArtifacts/CopyPublishBuildArtifacts.ps1 (hunk 1 above)
Comment (vishnugms) [inline Tasks/CopyPublishBuildArtifacts/CopyPublishBuildArtifacts.ps1]: This check and message reads task to be more appropriate to build, instead of saying where this task is not supported, I feel it's better to say where it is supported in case user sees this error..

File: Tasks/PublishBuildArtifacts/publishbuildartifacts.ts (hunk 2 above)
Comment (vishnugms) [inline Tasks/PublishBuildArtifacts/publishbuildartifacts.ts]: true. adding these.

File: Tasks/PublishBuildArtifacts/publishbuildartifacts.ts (hunk 3 above)
Comment (vishnugms) [inline Tasks/PublishBuildArtifacts/publishbuildartifacts.ts]: thanks for catching. removed it.

File: Tasks/PublishBuildArtifacts/publishbuildartifacts.ts (hunk 2 above)
Comment (vishnugms) [inline Tasks/PublishBuildArtifacts/publishbuildartifacts.ts]: Eric, this .js file is not added for these as well as other tasks beforel. Are you sure we want to add it?
